autogpt/core/messaging/base.py

```diff
@@ -1,73 +1,193 @@
-from abc import ABC, abstractmethod
-from enum import Enum, StrEnum
-import dataclasses
-from typing import Any, List, Callable, Union
-
-
-class MessageCategory(StrEnum):
-    COMMAND = "command"
-    AGENT_INSTANTIATION = "agent_instantiation"
-    USER_INPUT = "user_input"
-    USER_PROMPT = "user_prompt"
-    AGENT_MESSAGE = "agent_message"
-    SELF_FEEDBACK = "self_feedback"
-    PLAN = "plan"
+"""Abstract interface for a message broker system."""
+import abc
+from dataclasses import dataclass
+from datetime import datetime
+from enum import StrEnum
+from typing import Callable, Dict
 
 
 class Role(StrEnum):
+    """Enumeration of all entities capable of sending messages"""
+
     USER = "user"
-    SYSTEM = "system"
-    ASSISTANT = "assistant"
+    CLIENT_APPLICATION = "client_application"
+    AGENT_FACTORY = "agent_factory"
+    AGENT = "agent"
 
 
```

Comment:
# add dataclass for message priority
@dataclass
class Priority:
    """Struct for message priority."""
    # Some way to describe importance of the message for the current process
    value: str

    LOW = "low"
    NORMAL = "normal"
    HIGH = "high"
    CRITICAL = "critical"

    def __str__(self):
        return self.value